src/main/java/views/generate_summary_views/GenerateSummaryP.java

```diff
@@ -0,0 +1,104 @@
+package views.generate_summary_views;
+
+import org.jfree.chart.ChartFactory;
+import org.jfree.chart.ChartPanel;
+import org.jfree.chart.JFreeChart;
+import javax.swing.*;
+
+import org.jfree.chart.plot.CategoryPlot;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.chart.renderer.category.BarRenderer;
+import org.jfree.chart.title.TextTitle;
+import org.jfree.chart.title.Title;
+import org.jfree.data.category.CategoryDataset;
+import org.jfree.data.category.DefaultCategoryDataset;
+
+import org.jfree.data.general.DefaultPieDataset;
+import org.jfree.data.general.PieDataset;
+import use_cases.generate_summary_use_case.GenerateSummaryOB;
+import use_cases.generate_summary_use_case.GenerateSummaryOD;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * A class that generates the Bar Graph and Pie Chart given the GenerateSummaryOD outputData.
+ * Code for Bar Graph inspired by <a href="http://www.java2s.com/Code/Java/Chart/JFreeChartStackedBarChartDemo4.htm">this website.</a>
+ * Code for Pie Chart inspired by <a href="https://www.tutorialspoint.com/jfreechart/jfreechart_pie_chart.htm">this website.</a>
+ */
+public class GenerateSummaryP extends JFrame implements GenerateSummaryOB {
+    private double remainder;
+    private Map<String, ArrayList<Double>> statisticalData;
+
+    /**
+     * Generates a graphical representation of the formatted data associated with the current month and modifies the
+     * outputData to contain the resulting JPanels.
+     * @param outputData a GenerateSummaryOD object holding the formatted data needed to generate the graphs
+     */
+    @Override
+    public void formatOutputData(GenerateSummaryOD outputData) {
+        this.remainder = outputData.getRemainder();
+        this.statisticalData = outputData.getStatisticalData();
+        outputData.setBarGraphPanel(generateBarGraph());
+        outputData.setPieChartPanel(generatePieChart());
+    }
```

Comment:
This method should return GenerateSummaryOD object, which will be passed all the way to your view. Your view can have the methods below as private helper methods, and it will need a frame to hold all outputs from these methods.

Reply:
Okay, that will be in my next commit when I refactor the code! Thanks for figuring out the issue :).